solarforecastarbiter/metrics/calculator.py

```diff
@@ -206,3 +215,82 @@ def calculate_deterministic_metrics(processed_fx_obs, categories, metrics,
     out['values'] = tuple(metric_vals)
     calc_metrics = datamodel.MetricResult.from_dict(out)
     return calc_metrics
+
+
+def calculate_event_metrics(proc_fx_obs, categories, metrics):
```

Comment:
A lot of copy/paste in this function. One approach could be to be to refactor the common parts into smaller private functions that `calculate_deterministic_metrics` and `calculate_event_metrics` both call. What do you think?

Reply:
Thanks for suggesting that. I agree there's a lot of redundancy that could be refactored into smaller private functions for the deterministic and event metrics (and likely the probabilistic metrics as well). I'll work on revising.

Reply:
@wholmgren I just realized I have not addressed this refactoring suggestion yet. If it's still essential for this PR, I'll work on that today.

Reply:
ok to leave this alone